### PR TITLE
[BEAM-6569] Re-add gcsio_test to py3-gcp test suite

### PR DESCRIPTION
**Pull request by RobbeSneyders**
This PR re-adds the gcio_test to the py3-gcp tox suite. This test was removed in #7691 due to being flaky.

However, the problem here was not a flaky test, but that the test was included twice in the test suite (see [this comment](https://github.com/apache/beam/pull/7691#issuecomment-459431122)).
The same mocked object is used twice, which resulted in the iterator being exhausted during the second test.

In [the logs](https://builds.apache.org/job/beam_PreCommit_Python_Commit/4034/console) of one of the runs that flagged this test as flaky, we can see that the test passed the first time, and failed the second time.

> 12:54:06 test_rename (apache_beam.io.gcp.gcsio_test.TestGCSIO) ... ok
> ...
> 12:54:06 test_rename (apache_beam.io.gcp.gcsio_test.TestGCSIO) ... ERROR

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by RobbeSneyders** [comment]
R: @tvalentyn @aaltay

File: sdks/python/tox.ini [hunk 1]
```diff
@@ -58,7 +58,7 @@ setenv =
...
...cs,apache_beam.portability,apache_beam.pipeline_test,apache_beam.pvalue_test,apache_beam.runners,apache_beam.io.hadoopfilesystem_test,apache_beam.io.hdfs_integration_test,apache_beam.io.gcp.tests.utils_test,apache_beam.io.gcp.big_query_query_to_table_it_test,apache_beam.io.gcp.bigquery_io_read_it_test,apache_beam.io.gcp.bigquery_test,apache_beam.io.gcp.gcsfilesystem_test,apache_beam.io.gcp.pubsub_integration_test,apache_beam.io.hdfs_integration_test,apache_beam.io.gcp.internal,apache_beam.io.filesystem_test,apache_beam.io.filesystems_test,apache_beam.io.range_trackers_test,apache_beam.io.sources_test,apache_beam.transforms,apache_beam.testing,apache_beam.io.filesystemio_test,apache_beam.io.localfilesystem_test,apache_beam.io.range_trackers_test,apache_beam.io.restriction_trackers_test,apache_beam.io.source_test_utils_test,apache_beam.io.concat_source_test,apache_beam.io.filebasedsink_test,apache_beam.io.filebasedsource_test,apache_beam.io.textio_test,apache_beam.io.tfrecordio_test,apache_beam.examples.wordcount_debugging_test,apache_beam.examples.wordcount_minimal_test,apache_beam.examples.wordcount_test,apache_beam.io.parquetio_test,apache_beam.examples.cookbook.bigtableio_it_test
```
Comment (tvalentyn) [inline sdks/python/tox.ini]: I suggest to omit the change in line 61 to avoid merge conflict with https://github.com/apache/beam/pull/7644/files.

Comment (RobbeSneyders) [reply sdks/python/tox.ini]: Good catch, thanks!
Updated

**Comment by tvalentyn** [comment]
Run Python PreCommit

**Comment by tvalentyn** [comment]
Run Python PreCommit

**Comment by tvalentyn** [comment]
Run Python PreCommit

**Comment by RobbeSneyders** [comment]
Thanks for testing this. I couldn't get it to fail locally. I'll look further into it, taking your comment on [BEAM-6594](https://issues.apache.org/jira/browse/BEAM-6569?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=16760658#comment-16760658) into account.

**Comment by Juta** [comment]
Run Python PreCommit

**Comment by RobbeSneyders** [comment]
Retest this please

**Comment by RobbeSneyders** [comment]
These changes were added in another PR.

**Pull request closed without merging**